src/IO/S3/Client.cpp

```diff
@@ -379,10 +385,10 @@ Model::HeadObjectOutcome Client::HeadObject(HeadObjectRequest & request) const
 
     request.overrideURI(std::move(*bucket_uri));
 
-    /// The next call is NOT a recurcive call
-    /// This is a virtuall call Aws::S3::S3Client::HeadObject(const Model::HeadObjectRequest&)
-    return enrichErrorMessage(
-        HeadObject(static_cast<const Model::HeadObjectRequest&>(request)));
+    if (isClientForDisk())
+        CurrentMetrics::add(CurrentMetrics::S3DiskNoKeyErrors);
+
+    return enrichErrorMessage(std::move(result));
```

Comment:
Are you sure here?

Reply:
That is very strange that there is no sensitive test for that in our CI.

Reply:
I think the call is missed
```
HeadObject(static_cast<const Model::HeadObjectRequest&>(request))
```

Reply:
agreed, fixed this